Sort toolings with Db

ViacheslavUdalov opened the pull request:
![photo_2024-04-25_22-36-01](https://github.com/user-attachments/assets/6fc112c3-52f5-4f5e-a56a-4cfd7b259c2e)

ViacheslavUdalov committed:
longest common substring, greatest increasing substring, levinstein.

ViacheslavUdalov committed:
алгоритм Кнута-Морриса-Пратта

ViacheslavUdalov committed:
выводятся данные на страничку, все сортировки, сохранены в бд.

ViacheslavUdalov committed:
туллинги для всех сортировок

ViacheslavUdalov committed:
фикс тулингов

ViacheslavUdalov committed:
recreateBdData для bubbleSort, обновляет данные времени по типу массива и записывает в бд.
Пример использования
node bubbleSort.js -t 1000 -a s -bd recreate

MrMedved commented on public/sorts/bubbleSort.js:
```diff
@@ -0,0 +1,71 @@
+    console.log(timeEnd);
+    // console.log(timeConsole)
+
+    switch (options.bdProcess) {
+        case 'recreate' :
+            await recreateResultDB('bubble', parseInt(options.bubble), arrayType, timeEnd)
+            break;
+        default:
+            console.log('default Bd');
+            await recreateResultDB('bubble', parseInt(options.bubble), arrayType, timeEnd)
+            break;
+    }
+}
```
А опцию не писть в дб дадут?

## По джобам
- почистить дб (для этого типа только)
- записать ответ в дб
- почистить и записать
- посмотреть то, что уже есть (вывести построчно сейчас)
- хотя сделай сразу и вариант жисоном вернуть

Для ясности: если флага нет, то с базой ничего делать вообще не надо

MrMedved commented on dbInteraction/recreateSortType.js:
```diff
@@ -0,0 +1,36 @@
+export async function recreateResultDB(sortType, arraySize, arrayType, time) {
+    try {
+        // получение данных по типу сортировки и размеру массива
+        let getData = await fetch(`http://localhost:3000/results?sortType=${sortType}&arraySize=${arraySize}`);
```
Если реассайнить не будешь, то `const`

MrMedved commented on dbInteraction/recreateSortType.js:
```diff
@@ -0,0 +1,36 @@
+        }
+        let dataForMap = await getData.json();
+
+        // по нулевому индексу, потому что в бд есть дупликат
```
Ну сделай для этого еще переменную, важно чтобы читаемый код был. Типа `resultItem.times[arrayType] = time` понятнее будет

MrMedved commented on dbInteraction/recreateSortType.js:
```diff
@@ -0,0 +1,36 @@
+        console.log("new getData")
+        console.log(dataForMap);
+
+        // обноление по айди
```
*обновление (описка)

MrMedved commented on dbInteraction/recreateSortType.js:
```diff
@@ -0,0 +1,36 @@
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                times: getData.times
```
донгл

MrMedved replied:
date: new Date().toISOString()
 
 и тут донгл

ViacheslavUdalov replied:
что значит донгл

MrMedved replied:
Висящая запятая. Запятая у последнего элемента в массиве. Когда стопкой их пишешь, надо чтобы была

MrMedved reviewed (changes requested):
Сделай еще флаг -в --вербос, и чтобы консоль логи только с ним выводились, а по дефолту чтобы только то что мы в флаге попросили/ничего. Это чтобы мы потом это поюзали в админке.

Еще пара комментов по дебешке, в остальном ништяк👍

ViacheslavUdalov committed:
создание одной сортировки и сохранение в бд по значению размера массива и типу сортировки.
рефактор recreateSortType.js, в функции создания новой сортировки, проверка на наличии уже в бд сортировки по таким данным.
Вывод времени в консоль если есть флаг -v

ViacheslavUdalov commented:
сделал, вотс некст

MrMedved commented on public/sorts/bubbleSort.js:
```diff
@@ -1,17 +1,21 @@
 const options = program.opts();
 
 if (options.bubble) {
+    console.log(options.consoleWriting)
```
А это же напишется, даже если не просили)

ViacheslavUdalov replied:
запятые поставил, вот так получилось
  'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                times: getData.times,
                Верно? запятая в самом конце

ViacheslavUdalov replied:
а, ну да, я это прям перед самым коммитом написал, для теста, забыл сделать правильно, щас сделаю

MrMedved commented on public/sorts/bubbleSort.js:
```diff
@@ -41,16 +45,26 @@ if (options.bubble) {
-            await recreateResultDB('bubble', parseInt(options.bubble), arrayType, timeEnd)
+            if (options.consoleWriting) {
+                console.log('default Bd');
+            }
```
В эти функции тоже надо будет передавать этот вербос флаг, чтобы они тоже молча работу делали)

ViacheslavUdalov replied:
ок ща сделаю

ViacheslavUdalov replied:
сделано

MrMedved reviewed (approved):
Ща принимаем чтобы дальше ехать, но ты посмотри комменты

ViacheslavUdalov committed:
добавление в функции работы с бд - вывод в коносоль если есть флаг и нет вывода если флага нет
